src/Components/Patient/PatientPrivacyToggle.tsx

```diff
@@ -0,0 +1,153 @@
+import * as Notification from "../../Utils/Notifications.js";
+import CareIcon from "../../CAREUI/icons/CareIcon.js";
+import { ConsultationModel } from "../Facility/models.js";
+import { useDispatch } from "react-redux";
+import { useEffect, useState } from "react";
+import { UserRole } from "../../Common/constants.js";
+import {
+  getConsultationBed,
+  togglePatientPrivacy,
+} from "../../Redux/actions.js";
+import useAuthUser from "../../Common/hooks/useAuthUser.js";
+interface PatientPrivacyToggleProps {
+  consultationId: string;
+  consultation?: ConsultationModel | null;
+  fetchPatientData?: (state: { aborted: boolean }) => void;
+}
+export default function PatientPrivacyToggle(props: PatientPrivacyToggleProps) {
+  const { consultationId, consultation, fetchPatientData } = props;
+  const [privacy, setPrivacy] = useState<boolean>(false);
+  const user = useAuthUser();
+  const dispatch: any = useDispatch();
+
+  //condititonally render the privacy toggle button depending on user role
+  const allowPrivacyToggle = () => {
+    const currentUserType: UserRole = user.user_type;
+    if (
+      currentUserType == "DistrictAdmin" ||
+      currentUserType == "StateAdmin" ||
+      currentUserType == "LocalBodyAdmin" ||
+      (currentUserType == "Doctor" &&
+        user?.home_facility_object?.id === consultation?.facility) ||
+      (currentUserType == "Staff" &&
+        user?.home_facility_object?.id === consultation?.facility) ||
+      currentUserType == "WardAdmin"
+    )
+      return true;
+
+    return false;
+  };
+  //hook to fetch the privacy info of the patient
+  useEffect(() => {
+    const getPrivacyInfo = async () => {
+      if (
+        consultation?.current_bed?.privacy == true ||
+        consultation?.current_bed?.privacy == false
+      ) {
+        setPrivacy(consultation?.current_bed?.privacy);
+        return;
+      }
+      const bedId = consultation?.current_bed?.bed_object?.id;
+      const consultationBedID = consultation?.current_bed?.id;
+      try {
+        const res = await dispatch(
+          getConsultationBed(
+            consultationId,
+            bedId as string,
+            consultationBedID as string
+          )
+        );
+        if (
+          res &&
+          res.status === 200 &&
+          res?.data &&
+          (res.data?.privacy == true || res.data?.privacy == false)
```

Comment:
same here, if you are checking weather the ...privacy is boolean or not use typeof ...privacy === "boolean"

Also you are using == which means loose comparison, so the above check is always true